megamek/src/megamek/common/Tank.java

```diff
@@ -83,6 +84,13 @@ public class Tank extends Entity {
     public static final int CRIT_TURRET_LOCK = 13;
     public static final int CRIT_TURRET_DESTROYED = 14;
 
+    //Fortify terrain just like infantry
```

Comment:
Since this copies the infantry code (mostly) I suggest removing it from both and placing it into Entity unless there is some good reason against that. Entity can respond with 0 or false to all such calls, overridden in Inf&Tank. This removes the need for class casts when calling dugIn().

Reply:
This includes the methods added to Tank of course

Reply:
So copy out the getter and setter along with the variables? Then add Override to the Infantry and Tank functions?

Reply:
So I tried that and I run into other problems with class casts being intertwined within those blocks from other sections of code. This is something that may just require a complete cleanup as another separate PR, probably by someone with more Java knowledge than me considering how old the code is. The only section that I can quickly get in place will be for Tank, else I will never find an end to this rabbit hole.

Plus it becomes, how much do we want to bloat entity to avoid class casts. For now my vote is to leave it alone.

Reply:
OK, then let's leave it. Entity is too big that is true, but the solution shouldn't be not using proper overrides, also it's nice not to have to use class casts. But that's for another cleanup then.